packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -103,124 +349,117 @@ const makeSwapLockMessages = (
 export const openPortfolio = (async (
   orch: Orchestrator,
   ctx: {
+    chainHubTools: {
+      getDenom: (brand: Brand) => Denom | undefined;
+    };
     zoeTools: GuestInterface<ZoeTools>;
     makePortfolioKit: () => PortfolioKit;
-    contract: EVMContractAddresses;
+    axelarChainsMap: AxelarChainsMap;
+    contractAddresses: EVMContractAddresses;
     inertSubscriber: GuestInterface<ResolvedPublicTopic<never>['subscriber']>;
   },
   seat: ZCFSeat,
   offerArgs: EVMOfferArgs, // TODO: USDN/USDC ratio
   // passed as a promise to alleviate contract start-up sync constraints
   localP: Promise<OrchestrationAccount<{ chainId: 'agoric-any' }>>,
 ) => {
+  mustMatch(offerArgs, EVMOfferArgsShape);
   await null; // see https://github.com/Agoric/agoric-sdk/wiki/No-Nested-Await
   try {
-    const { makePortfolioKit, contract } = ctx;
+    const {
+      makePortfolioKit,
+      contractAddresses,
+      axelarChainsMap,
+      chainHubTools,
+    } = ctx;
+    const { give } = seat.getProposal() as ProposalShapes['openPortfolio'];
     const kit = makePortfolioKit();
 
-    const initRemoteEVMAccount = async (protocol: YieldProtocol) => {
+    const initRemoteEVMAccount = async (amount: Amount<'nat'>) => {
```

Comment:
Would you please hoist the functions from inside `openPortfolio` like in #11505 ?

I tried to merge our branches, and this made it really challenging.